scrapers_next/ak/people.py

```diff
@@ -1,69 +1,143 @@
-from spatula import XmlListPage, URL, XPath
+import re
+import attr
+from spatula import HtmlPage, HtmlListPage, XPath, CSS, SelectorError
 from openstates.models import ScrapePerson
 
 
-ELEMENTS = (
-    "FirstName",
-    "MiddleName",
-    "LastName",
-    "EMail",
-    "Phone",
-    "District",
-    "Party",
-    "Building",
-    "Room",
-)
+@attr.s(auto_attribs=True)
+class PartialMember:
+    url: str
+    chamber: str = ""
 
 
-def _get_if_exists(item, elem):
-    val = item.xpath(f"./{elem}/text()")
-    if val:
-        return str(val[0])
+class LegDetail(HtmlPage):
+    example_source = "http://www.akleg.gov/basis/Member/Detail/32?code=BCH"
 
+    def process_page(self):
 
-class Legislators(XmlListPage):
-    session_num = "32"
-    source = URL(
-        "http://www.legis.state.ak.us/publicservice/basis/members"
-        f"?minifyresult=false&session={session_num}",
-        headers={"X-Alaska-Legislature-Basis-Version": "1.4"},
-    )
-    selector = XPath("//Member/MemberDetails")
+        details_div = CSS(".bioright").match(self.root)[0]
 
-    def process_item(self, item):
-        item_dict = {elem: _get_if_exists(item, elem) for elem in ELEMENTS}
-        chamber = item.attrib["chamber"]
-        code = item.attrib["code"].lower()
-
-        party = item_dict["Party"]
-        if party == "N":
-            party = "Independent"
-
-        person = ScrapePerson(
-            name="{FirstName} {LastName}".format(**item_dict),
-            given_name=item_dict["FirstName"],
-            family_name=item_dict["LastName"],
+        name_span = CSS(".formal_name").match(details_div)[0].text_content()
+        name_list = name_span.split(" ")
+        given_name = name_list[1]
+        family_name = " ".join(name_list[2:])
+
+        email = CSS("a").match(details_div)[0].text_content().strip()
+
+        div_text = details_div.text_content().replace("\r\n", " ")
```

Comment:
May want to `split` up the text on new lines so that it's easier to get text for specific lines rather than saying exactly how many characters should be captured on match (thinking phone numbers). Another option is to create separate arrays for each contact block since they have two `<br>` between to remove some of the similar code/sanity checks that happen further down.

Reply:
Re: our earlier conversation, I still chose to handle using text_content() for the whole div for now, in case AK's future web developer brings page more current (i.e. removes `<br>` tags).